packages/app/src/elements/audit.ts

```diff
@@ -0,0 +1,214 @@
+                        <pl-audit-list-item
+                            .listItems=${_reusedPasswords}
+                            .label=${$l("Reused Passwords")}
+                            .routeParams=${{ auditReused: "true" }}
+                        ></pl-audit-list-item>
+                        <pl-audit-list-item
+                            .listItems=${_weakPasswords}
+                            .label=${$l("Weak Passwords")}
+                            .routeParams=${{ auditWeak: "true" }}
+                        ></pl-audit-list-item>
+                        <pl-audit-list-item
+                            .listItems=${_compromisedPasswords}
+                            .label=${$l("Compromised Passwords")}
+                            .routeParams=${{ auditCompromised: "true" }}
+                        ></pl-audit-list-item>
+                    </div>
+                </div>
+            </div>
+        `;
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        this.addEventListener("change", () => this.audit());
+    }
+
+    async audit() {
+        this._reusedPasswords = [];
+        this._weakPasswords = [];
+        this._compromisedPasswords = [];
+
+        const { vaults } = this.state;
+
+        // This prevents the same item from being considered more than once for different sections/audits
+        const checkedItemIds: Set<string> = new Set();
+
+        for (const vault of vaults) {
+            for (const item of vault.items) {
+                if (checkedItemIds.has(item.id)) {
+                    continue;
+                }
+
+                // TODO: Skip/ignore if the item doesn't have a password field
+
+                // TODO: Actually check if the audit is correct, instead of this random calculation
+                const randomNumber = Math.floor(Math.random() * 2);
+
+                if (randomNumber % 5 === 0) {
+                    this._reusedPasswords.push({
+                        vault,
+                        item,
+                    });
+
+                    checkedItemIds.add(item.id);
+                } else if (randomNumber % 5 === 1) {
+                    this._weakPasswords.push({
+                        vault,
+                        item,
+                    });
+
+                    checkedItemIds.add(item.id);
+                } else if (randomNumber % 5 === 2) {
+                    this._compromisedPasswords.push({
+                        vault,
+                        item,
+                    });
+
+                    checkedItemIds.add(item.id);
+                }
+
+                // TODO: Store + save (+ sync) last audit date and audit match booleans in items
+            }
+        }
+
+        this._countElements.forEach((countElement) => {
+            animateElement(countElement, { animation: "bounce" });
+        });
+
+        // TODO: This doesn't work, and shouldn't be necessary
+        this.stateChanged();
```

Comment:
Nope, I've never used this method and don't think this is meant to be called explicitly.